packages/core/src/lib/compactObject.ts

```diff
@@ -1,9 +1,7 @@
-export const compactObject = (obj: Record<string, unknown>) => {
-  const newObj: Record<string, unknown> = {}
-  Object.keys(obj).forEach((key) => {
-    const value = obj[key]
-    if (value !== undefined) newObj[key] = value
-  })
-
-  return newObj
+export const compactObject = (
+  obj: Record<string, unknown | undefined>,
+): Record<string, unknown> => {
+  return Object.fromEntries(
+    Object.entries(obj).filter(([_, value]) => value !== undefined),
+  )
```

Comment:
is this changing behaviour? this is used in many places

Reply:
Nope, same behaviour. Could just undo the change, just felt this was more elegant :)